python/ql/test/experimental/meta/ConceptsTest.qll

```diff
@@ -33,19 +33,41 @@ class SystemCommandExecutionTest extends InlineExpectationsTest {
   }
 }
 
-class DeserializationSinkTest extends InlineExpectationsTest {
-  DeserializationSinkTest() { this = "DeserializationSinkTest" }
+class UnmarshalingFunctionTest extends InlineExpectationsTest {
+  UnmarshalingFunctionTest() { this = "UnmarshalingFunctionTest" }
 
   override string getARelevantTag() { result = "getData" }
 
   override predicate hasActualResult(Location location, string element, string tag, string value) {
-    exists(DeserializationSink ds, DataFlow::Node data |
-      exists(location.getFile().getRelativePath()) and
-      data = ds.getData() and
-      location = data.getLocation() and
-      element = data.toString() and
-      value = value_from_expr(data.asExpr()) and
-      tag = "getData"
+    exists(location.getFile().getRelativePath()) and
+    exists(UnmarshalingFunction ds, string unsafe |
+      (
+        ds.unsafe() and unsafe = "UNSAFE_"
+        or
+        not ds.unsafe() and unsafe = ""
+      ) and
```

Comment:
I'm not a big fan of this way of using tags. Could we instead use something like the following scheme

```py
mylib.deserizlie(input_string) # $deserializeInput=input_string $deserializeOutput=Attribute() $deserializeFormat="mylib" $deserializeUnsafe
```

This way we can annotate safe and unsafe deserialization in the same way, and just mark unsafe ones with a special tag.

Although it does makes the name longer, I do think we need the `deserialize` prefix. Once we also model serialization, we will probably need to talk about their inputs/outputs as well, and then we get a naming conflict :neutral_face:

Reply:
Done